packages/ui/src/EmptyState.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FC } from 'react'
+import cn from 'classnames'
+import { ChevronRight } from 'react-feather'
+
+import { ButtonTypeAnchorProps, ButtonProps, Button } from './Button'
+import { Link } from './Link'
+import { Icon, IconProps } from './Icon'
+
+import styles from './EmptyState.module.scss'
+
+type EmptyStateSize = 'normal' | 'large'
+
+type EmptyStateActionButton = {
+  direction?: 'vertical'
+  actionOnClick: ButtonProps['onClick']
+  actionHref?: never
+  actionTarget?: never
+  actionRel?: never
+} & {
+  size?: EmptyStateSize
+}
+
+type EmptyStateActionLink = {
+  direction: 'horizontal'
+  actionHref: ButtonTypeAnchorProps['href']
+  actionTarget?: ButtonTypeAnchorProps['target']
+  actionRel?: ButtonTypeAnchorProps['rel']
+  actionOnClick?: never
+} & {
+  size?: never
+}
+
+type EmptyStateAction = {
+  action: string
```

Comment:
Should we make `action` optional? We might want to show an empty state without any action, for example in tables.